descartes_moveit/src/seed_search.cpp

```diff
@@ -0,0 +1,290 @@
+#include <descartes_moveit/seed_search.h>
+
+#include <ros/ros.h>
+
+using namespace descartes_moveit;
+
+typedef std::vector<unsigned> BijectionVec;
+typedef std::pair<unsigned, unsigned> JointPair;
+typedef std::vector<JointPair> JointPairVec;
+typedef std::vector<double> JointConfig;
+typedef std::vector<JointConfig> JointConfigVec;
+
+
+/**
+ * @brief Forward kinematics helper
+ */
+bool doFK(moveit::core::RobotState& state,
+          const moveit::core::JointModelGroup* group,
+          const std::string& tool,
+          const JointConfig& joint_pose,
+          Eigen::Affine3d& result)
+{
+  state.setJointGroupPositions(group, joint_pose);
+  if (!state.knowsFrameTransform(tool))
+  {
+    ROS_WARN("No transform to this tool frame");
+    return false;
+  }
+
+  if (!state.satisfiesBounds())
+  {
+    ROS_WARN("Joint angles do not satisfy robot bounds");
+    return false;
+  }
+
+  result = state.getFrameTransform(tool);
+  return true;
+}
+
+/**
+ * @brief Inverse kinematics helper. Returns the solution in the result parameter.
+ *        May return false if IK failed.
+ */
+bool doIK(moveit::core::RobotState& state,
+          const moveit::core::JointModelGroup* group,
+          const std::string& group_name,
+          const std::string& tool,
+          const Eigen::Affine3d& pose,
+          const JointConfig& seed,
+          JointConfig& result)
+{
+  const static int N_ATTEMPTS = 1;
+  const static double IK_TIMEOUT = 0.01;
+
+  state.setJointGroupPositions(group_name, seed);
+  if (!state.setFromIK(group, pose, tool, N_ATTEMPTS, IK_TIMEOUT))
+  {
+    return false;
+  }
+  state.copyJointGroupPositions(group, result);
+  return true;
+}
+
+/**
+ * @brief Returns true if the determinant of the jacobian is not zero.
```

Comment:
Is the logic correct there.  It returns true if the determinant is not zero...i.e. NOT singular.